src/pages/rate/[appointment_id].tsx

```diff
@@ -0,0 +1,240 @@
+    return;
+  }
+  if (!userStatus.user) {
+    router.push(PAGE_PATHS.LOGIN);
+    return;
+  }
+  return (
+    <>
+      <Navbar />
+
+      <IconButton
+        onClick={backToMyAppointments}
+        style={{ position: "absolute", top: 86, left: 20, zIndex: "1" }}
+      >
+        <ArrowBackIcon fontSize="large" color="secondary" />
+      </IconButton>
+
+      <Stack spacing={4} sx={{ marginBottom: "2vh" }} alignItems="center">
+        {/* Page header */}
+        <Box sx={{ marginTop: "3vh" }}>
+          <Typography variant="h1">Rate Appointment</Typography>
+        </Box>
+
+        <Stack spacing={5} direction="row">
+          <Card sx={ParticipantAppointmentStyle.Card}>
+            <Stack spacing={0} alignItems="center" justifyContent="center">
+              {/* Post title */}
+              <Box style={ParticipantAppointmentStyle.TextField}>
+                <TitleTextField
+                  name="title"
+                  header="Title"
+                  placeholder="This is Post Title"
+                  value={appointment.detailHeader.title}
+                  handleValueChange={() => {}}
+                  isErr={false}
+                  errMsg=""
+                  readOnly={true}
+                />
+              </Box>
+
+              {/* Location */}
+              <Box sx={ParticipantAppointmentStyle.TextField}>
+                <LocationTextField
+                  header="Location"
+                  placeholder="Enter Location"
+                  initialValue={appointment.detailHeader.location}
+                  onChange={() => {}}
+                  isErr={false}
+                  errMsg=""
+                  readOnly={true}
+                />
+              </Box>
+
+              {/* Date & Time */}
+              <Box sx={ParticipantAppointmentStyle.TextField}>
+                <DisplayDateTime
+                  header="Date & Time"
+                  value={`${appointment.detailHeader.startDateTime} - ${appointment.detailHeader.endDateTime}`}
+                  readOnly={true}
+                />
+              </Box>
+
+              {/* Tags */}
+              <Box sx={ParticipantAppointmentStyle.TextField}>
+                <Tags
+                  header="Tag"
+                  value={appointment.detailHeader.tags.map((t, index): Tag => {
+                    return { name: t, id: index };
+                  })}
+                  handleValueChange={() => {}}
+                  menuValue={[]}
+                  isErr={false}
+                  errMsg=""
+                  readOnly={true}
+                />
+                <FormHelperText>{"\u00A0"}</FormHelperText>
+              </Box>
+
+              {/* Description */}
+              <Box sx={ParticipantAppointmentStyle.TextField}>
+                <DescriptionTextField
+                  name="description"
+                  header="Description"
+                  placeholder="Enter Description Here"
```

Comment:
"Enter Description Here" แต่แก้ไขไม่ได้

Reply:
จริง ๆ description ใน post มันบังคับใส่ตอนสร้าง แต่เข้าใจว่า appointment ที่ mock ขึ้นมาไม่ได้ใส่มันเลยขึ้น placeholder แทน

Reply:
If we can't assume that data from backend is correct and error occur like missing title/location/description/date-time what should the page look like?